app/models/ssh_execution_provider.rb

```diff
@@ -25,6 +25,20 @@ def ssh_key_passphrase(host)
       host_setting(host, :remote_execution_ssh_key_passphrase)
     end
 
+    def ssh_params(host)
+      proxy_selector = ::RemoteExecutionProxySelector.new
+      proxy = proxy_selector.determine_proxy(host, 'SSH')
+      {
+        :hostname => find_ip_or_hostname(host),
+        :proxy => proxy && proxy.url,
```

Comment:
`proxy` can also be a symbol like `:not_available` and then `proxy.url` crashes.